src/borg/testsuite/signature.py

```diff
@@ -0,0 +1,163 @@
+
+import pytest
+
+from ..signature import SignedFile, SignatureError
+
+
+class TestReadSignatures:
+    def test_no_signature(self, tmpdir):
+        protected_file = tmpdir.join('file')
+        protected_file.write('1234')
+        assert not SignedFile.read_signatures(str(protected_file), None)
+
+    def test_truncated_signature(self, tmpdir):
+        protected_file = tmpdir.join('file')
+        protected_file.write('1234')
+        tmpdir.join('file.signature').write('')
+        with pytest.raises(SignatureError):
+            SignedFile.read_signatures(str(protected_file), None)
+
+    def test_unknown_algorithm(self, tmpdir):
+        class SomeSigner:
+            NAME = 'HMAC_FOOHASH9000'
+        protected_file = tmpdir.join('file')
+        protected_file.write('1234')
+        tmpdir.join('file.signature').write('{"algorithm": "HMAC_SERIOUSHASH", "signature": "1234"}')
+        assert not SignedFile.read_signatures(str(protected_file), SomeSigner())
+
+    @pytest.mark.parametrize('json', (
+        '{"ALGORITHM": "HMAC_SERIOUSHASH", "signature": "1234"}',
+        '[]',
+        '1234.5',
+        '"A string"',
+        'Invalid JSON',
+    ))
+    def test_malformed(self, tmpdir, json):
+        protected_file = tmpdir.join('file')
+        protected_file.write('1234')
+        tmpdir.join('file.signature').write(json)
+        with pytest.raises(SignatureError):
+            SignedFile.read_signatures(str(protected_file), None)
+
+    def test_valid(self, tmpdir):
+        class SomeSigner:
+            NAME = 'HMAC_FOO1'
+
+        protected_file = tmpdir.join('file')
+        protected_file.write('1234')
+        tmpdir.join('file.signature').write('{"algorithm": "HMAC_FOO1", "signatures": {"final": "1234"}}')
+        assert SignedFile.read_signatures(str(protected_file), SomeSigner()) == {'final': '1234'}
+
+
+class TestSignedFile:
+    @pytest.fixture
+    def key(self):
+        return bytes(1234)
```

Comment:
any special reason to use 1234 zero bytes? Usually a hmac-sha512 key is 512bits (64 bytes).